filament/src/details/Texture.cpp

```diff
@@ -183,6 +183,12 @@ Texture* Texture::Builder::build(Engine& engine) {
         }
     }
 
+    // If this texture is a color attachment, then make sure it can be the source of a copy for
+    // readPixels().
+    if (any(mImpl->mUsage & TextureUsage::COLOR_ATTACHMENT)) {
+        mImpl->mUsage |= TextureUsage::BLIT_SRC;
+    }
```

Comment:
I think we should have a bug for this and have a plan to fix all the call sites and eventually remove this. I guess this is an API change: we need to document that `readPixels()` (and maybe others) need the `BLIT_SRC` usage bit. And we also need to have a PRECONDITON check in readPixels() so that it crashes early instead of in the backends.

Reply:
Maybe we don't even need this?  It looks like `readPixels` only works on the default rendertarget: 
https://github.com/google/filament/blob/main/filament/src/details/Renderer.cpp#L451

Reply:
Updated with warning message and added API comment.